components/tools/OmeroPy/src/omero/util/metadata_mapannotations.py

```diff
@@ -58,18 +61,13 @@ def __init__(self, ma, primary_keys=None, unique_keys=True):
         except TypeError:
             mapvalue = []
         self.kvpairs, self.primary = self.process_keypairs(
-            mapvalue, primary_keys, unique_keys)
+            mapvalue, primary_keys)
         self.parents = set()
 
-    def process_keypairs(self, kvpairs, primary_keys, unique_keys):
+    def process_keypairs(self, kvpairs, primary_keys):
         if len(set(kvpairs)) != len(kvpairs):
             raise ValueError('Duplicate key-value pairs found: %s' % kvpairs)
```

Comment:
@eleanorwilliams ran into a problem with idr0018 where there are duplicate name-values within the same MapAnnotation. Should this be allowed?